pandas/tests/test_frame.py

```diff
@@ -16064,6 +16064,18 @@ def bar(self):
         with tm.assertRaisesRegexp(AttributeError, '.*i_dont_exist.*'):
             A().bar
 
+    def test_multithread_stringio_read_csv(self):
+        from io import BytesIO
+        from multiprocessing.pool import ThreadPool
+
+        bytes = ['\n'.join(['%d,%d,%d' % (i, i, i) for i in range(10000)]).encode()
+                 for j in range(100)]
+        files = [BytesIO(b) for b in bytes]
+
+        # Read all files in many threads
+        pool = ThreadPool(8)
```

Comment:
assert that the read in values match a single threaded reader. (e.g. compare frames)